index.js

```diff
@@ -175,6 +178,18 @@ class RenderPDF {
         if(this.options.pageRanges !== undefined) {
             options.pageRanges = this.options.pageRanges;
         }
+      
+        if (this.options.displayHeaderFooter !== undefined) {
+            options.displayHeaderFooter = !!this.options.displayHeaderFooter;
+        }
+        
+        if (this.options.headerTemplate !== undefined) {
+            options.headerTemplate = this.options.headerTemplate;
+        }
+        
+        if (this.options.footerTemplate !== undefined) {
+            options.footerTemplate = this.options.footerTemplate;
+        }
```

Comment:
- What's happening if displayHeaderFooter, headerTemplate or footerTemplate are defined as "null".
- Could you replace the double !! to return the boolean value of 'this.options.displayHeaderFooter' and replace it with the following code.

``` javascript
       if (this.options.displayHeaderFooter) {
            options.displayHeaderFooter = Boolean(this.options.displayHeaderFooter);
        }
```

With the following code, you can apply all of my suggestions.

``` javascript
       if (this.options.displayHeaderFooter) {
            options.displayHeaderFooter = Boolean(this.options.displayHeaderFooter);
        }

        if (this.options.headerTemplate) {
            options.headerTemplate = this.options.headerTemplate;
        }

        if (this.options.footerTemplate) {
            options.footerTemplate = this.options.footerTemplate;
        }
```

But in my opinion, all files should be in Typescript to prevent these things.

Reply:
Thanks for review, I agree with your comments, but it seems the owner does not want this PR as it is open now for a few years. You're free to make the changes if you wish and use a forked version.

Reply:
> But in my opinion, all files should be in Typescript to prevent these things.
This project wasn't done in typescript only because of historical requirement that was purpose for creating this tool.
Currently it's no longer required, but I do not see value in doing rewrite